jax/scipy/linalg.py

```diff
@@ -264,3 +265,160 @@ def tril(m, k=0):
 @_wraps(scipy.linalg.triu)
 def triu(m, k=0):
   return np.triu(m, k)
+
+@custom_transforms
+def expm(A):
+    """
+    Compute the matrix exponential using Pade approximation.
+
+    Args:
+    A : square matrix of type float32, float64, complex64, complex128
+    
+    Returns:
+    Matrix exponential of A , square matrix of same dimensions as A
+        
+
+    See: "The Scaling and Squaring Method for the Matrix
+    Exponential Revisited", Nicholas Higham, 2005.
+
+    """
+    A = np.asarray(A)
+    A_L1 = np_linalg.norm(A,1)
+    n_squarings = 0
+    def _nest_where(vals, cases):
+      assert len(vals) == len(cases) - 1
+      if len(vals) == 1:
+        return np.where(
+            np.less(A_L1, vals[0]), cases[0], cases[1])
+      else:
+        return np.where(
+            np.less(A_L1, vals[0]), cases[0],
+            _nest_where(vals[1:], cases[1:]))
+    if A.dtype == 'float64' or A.dtype == 'complex128':
+       U3,V3 = _pade3(A)
+       U5,V5 = _pade5(A)
+       U7,V7 = _pade7(A)
+       U9,V9 = _pade9(A)
+       maxnorm = 5.371920351148152
+       n_squarings = np.maximum(0, np.floor_divide(np.log2(A_L1 / maxnorm),1))
+       A = A / 2**n_squarings
+       U13,V13 = _pade13(A)
+       conds=np.array([1.495585217958292e-002, 2.539398330063230e-001, 9.504178996162932e-001, 2.097847961257068e+000])
+       U=_nest_where(conds,(U3,U5,U7,U9,U13))
+       V=_nest_where(conds,(V3,V5,V7,V9,V13))
+    elif A.dtype == 'float32' or A.dtype == 'complex64':
+        U3,V3 = _pade3(A)
+        U5,V5 = _pade5(A)
+        maxnorm = 3.925724783138660
+        n_squarings = np.maximum(0, np.floor_divide(np.log2(A_L1 / maxnorm),1))
+        A = A / 2**n_squarings
+        U7,V7 = _pade7(A)
+        conds=np.array([4.258730016922831e-001, 1.880152677804762e+000])
+        U=_nest_where(conds,(U3,U5,U7))
+        V=_nest_where(conds,(V3,V5,V7))
+    else:
+        raise TypeError("A.dtype={} is not supported.".format(A.dtype))
+    P = U + V  # p_m(A) : numerator
+    Q = -U + V # q_m(A) : denominator
+    R = np_linalg.solve(Q,P)
+    # squaring step to undo scaling
+    def my_body_fun(i,R):
+      return np.dot(R,R)
+    lower = np.zeros(1, dtype=n_squarings.dtype)
+    lax.fori_loop(lower[0],n_squarings,my_body_fun,R)
```

Comment:
This line doesn't do anything -- `fori_loop` is purely function, so you need to use the return value.

Reply:
Okay. This should be fixed. My testing did not reveal the bug.